File: src/test/java/com/optimizer/handler/DraftKingsHandlerTest.java
package com.optimizer.handler;

import collect.DraftKingsData;
import com.optimizer.collect.MockResponses;
import handler.DraftKingsHandler;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import java.util.HashMap;
import java.util.Map;

import static org.mockito.Mockito.*;

class DraftKingsHandlerTest implements MockResponses {

    @Mock
    DraftKingsData draftKingsData;

    @InjectMocks
    DraftKingsHandler draftKingsHandler;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.initMocks(this);
    }

    @Test
    void shouldHandleRequestForDraftKingsData() {
        Map<String, String> testMap = new HashMap<>();
        testMap.put("sport", "nfl");
        draftKingsHandler.handleRequest(testMap);
        verify(draftKingsData).getAllContestData("nfl");
    }
}

File: src/test/kotlin/handler/DraftKingsHandlerTest.kt
package handler

import io.mockk.every
import io.mockk.spyk
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class DraftKingsHandlerTest {
    private val sport = "a sport"
    private val draftKingsHandler = spyk<DraftKingsHandler>()
    private val draftKingsResult: List<Map<String, Any>> = listOf()

    @BeforeEach
    fun setUp() {
        every { draftKingsHandler.getDraftKingsContestData(sport) } returns draftKingsResult
    }

    @Test
    fun `should handle request`() {
        val result = draftKingsHandler.handleRequest(mapOf("sport" to sport))
        assertEquals(draftKingsResult, result)
    }
}